src/Materials/customShaderCode.ts

```diff
@@ -0,0 +1,295 @@
+import { Effect } from "./effect";
+import { Mesh } from "../Meshes/mesh";
+import { Texture } from "./Textures/texture";
+import { MaterialDefines } from "./materialDefines";
+
+/**
+ * Defines a code piece to customize the shader.
+ */
+export class ShaderCustomPiece {

```

Comment:
should be in a separate folder with everything related to custom shader code (one file per entity)